### DISCUSSION TITLE
show difference of inherited and overwritten values on object

### DISCUSSION THREAD

**Discussion by langert** [pimcore/pimcore#14223] [Q&A / Support] [answered]
Hi,
with the attributes getter methods, I get the inherited values of an object attribute, fine! But is there a way to detect, if the value comes from the inherited parentObject or from the object variant itself?

**Comment by langert** [comment, marked as answer]
you have to fetch the object data with and without inheritance and then compare it

**Reply by shyam-raj-sampath** [reply]
Hi, I have been trying to fetch variant information with inherited data, but I am unable to do so. 

Can you please point to the code for the same? Any help will be really appreciated, as I have been stuck on this for a long time.

**Reply by langert** [reply]
In my case I do it like this:

```
$myObject = MyObject::getById($id);
$inheritedObject = $myObject->getNextParentForInheritance();

if($myObject->getMyField() != $inheritedObject->getMyField()){
  echo 'myObject has a custom not inherited value for myField';
}
```

**Reply by langert** [reply]
you could also use the method
`MyObject::setGetInheritedValues(false);`
get your values and reset
`MyObject::setGetInheritedValues(true);`
on your object. If you do so, you will get the not inherited values from your object directly an do any comparision

**Reply by shyam-raj-sampath** [reply]
Thanks a lot for all the help.. :)
